code/go/0chain.net/blobbercore/writemarker/worker.go

```diff
@@ -16,52 +16,80 @@ func SetupWorkers(ctx context.Context) {
 	go startRedeemWriteMarkers(ctx)
 }
 
-func RedeemMarkersForAllocation(ctx context.Context, allocationObj *allocation.Allocation) error {
-	rctx := datastore.GetStore().CreateTransaction(ctx)
-	db := datastore.GetStore().GetTransaction(rctx)
+func redeemWriterMarkersForAllocation(allocationObj *allocation.Allocation) {
 	defer func() {
-		err := db.Commit().Error
-		if err != nil {
-			Logger.Error("Error committing the writemarker redeem", zap.Error(err))
+		if r := recover(); r != nil {
+			logging.Logger.Error("[recover] redeemWriterMarkersForAllocation", zap.Any("err", r))
 		}
-		rctx.Done()
 	}()
 
-	writemarkers := make([]*WriteMarkerEntity, 0)
+	ctx := datastore.GetStore().CreateTransaction(context.TODO())
```

Comment:
@cnlangzi This `ctx` is used to store `db` client.
I suspect that this is a problem to use.

For example, if there were 10 writemarkers to redeem for an allocation. You redeemed 9 writemarkers and `db` transaction also has updated database accordingly.
At 10th writemarker, if any error occurs, it will rollback all the updated values which might not be desired. Next time it will again try to redeem already redeemed writemarker and it will never be able to redeem.

I think we should use separate db transaction for each writemarker updates so that once redeemed writemarker has `committed` status in database.

Reply:
Above `ctx` should be used only to update `allocationObj`. i.e. to update `isRedeemRequired`

Reply:
it happens on all blobber transaction and 0chain transaction whatever it is 1 or more transactions. I are not sure if 0chain transaction is idempotent. that is problem on all distributed transaction design. that is why we need it is idempotent, or 2-steps transactions.  cc @peterlimg 
I would suggest to leave it as it is now. 
A big issue should be raised to discuss how to keep its consistency on our distributed transaction design.

Reply:
@cnlangzi The other blobber transactions I think is a different story.
Here once a writemarker is redeemed, it should update its status independent of other write markers.

We just need to create new transaction and commit with it.
In readmarker I've done the same. Create new transaction for each readmarkers and commit/rollback with it only.

Reply:
let explain it more
- Redeemed WM on 0chain
- it is done on 0chain, but network is broken
- blobber gets a network error, and rollback db transaction

what happens on blobber side on next round?  do you think it is the same story ?

Reply:
anyway, updated code as your suggestion.